diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61d2cc6c5..215da8622 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -481,7 +481,7 @@ list(APPEND libopenmc_SOURCES
   src/external/Faddeeva.cc)
 
 # For Visual Studio compilers
-if(MSVC)
+if(MSVC OR MINGW)
   # Use static library (otherwise explicit symbol portings are needed)
   add_library(libopenmc STATIC ${libopenmc_SOURCES})
 
@@ -495,7 +495,7 @@ endif()
 add_library(OpenMC::libopenmc ALIAS libopenmc)
 
 # Avoid vs error lnk1149 :output filename matches input filename
-if(NOT MSVC)
+if(NOT (MSVC OR MINGW))
   set_target_properties(libopenmc PROPERTIES OUTPUT_NAME openmc)
 endif()
 
